tests/Unit/Parallel/Test_MemoryMonitor.cpp

```diff
@@ -24,7 +99,48 @@ void test_tags() {
   CHECK(subpath == "/MemoryMonitors/MockMemoryMonitor");
 }
 
+void test_initialize() {
+  INFO("Test InitializeMemoryMonitor");
+  using metavars = TestMetavariables;
+  using mem_mon_comp = MockMemoryMonitor<metavars>;
+  using sing_comp = SingletonParallelComponent<metavars>;
+  using group_comp = GroupParallelComponent<metavars>;
+  using array_comp = ArrayParallelComponent<metavars>;
+  using nodegroup_comp = NodegroupParallelComponent<metavars>;
+
+  ActionTesting::MockRuntimeSystem<metavars> runner{{}};
+  runner.set_phase(TestMetavariables::Phase::Initialization);
+
+  ActionTesting::emplace_singleton_component<sing_comp>(
+      make_not_null(&runner), ActionTesting::NodeId{0},
+      ActionTesting::LocalCoreId{0});
+  ActionTesting::emplace_group_component<group_comp>(make_not_null(&runner));
+  ActionTesting::emplace_nodegroup_component<nodegroup_comp>(
+      make_not_null(&runner));
+  ActionTesting::emplace_array_component<array_comp>(
+      make_not_null(&runner), ActionTesting::NodeId{0},
+      ActionTesting::LocalCoreId{0}, 0);
+
+  // Empty initial values
+  ActionTesting::emplace_singleton_component_and_initialize<mem_mon_comp>(
+      make_not_null(&runner), ActionTesting::NodeId{0},
+      ActionTesting::LocalCoreId{0}, {});
+
+  std::unordered_map<
+      std::string, std::unordered_map<double, std::unordered_map<int, double>>>
+      empty_holder{};
+
+  using tag = mem_monitor::Tags::MemoryHolder;
+  CHECK(ActionTesting::box_contains<mem_mon_comp, tag>(runner, 0));
+  CHECK(ActionTesting::tag_is_retrievable<mem_mon_comp, tag>(runner, 0));
```

Comment:
Make this a `REQUIRE` since the rest of the test won't give reasonable results if this doesn't pass?